In [1]:
# Import the necessary libraries
import pandas as pd

In [108]:
train = pd.read_csv('Train.csv')
test = pd.read_csv('Test.csv')
description = pd.read_csv('VariableDefinitions.csv')

## Data Understanding
- Data info
- Data shape
- Missing data
- Duplicated data

Description of the data in use

In [109]:
description

Variable Definitions  \
0                  country   
1                     year   
2                 uniqueid   
3            location_type   
4         cellphone_access   
5           household_size   
6        age_of_respondent   
7     gender_of_respondent   
8   relationship_with_head   
9           marital_status   
10         education_level   
11                job_type   

                                                                                                                                                                                                                                        Unnamed: 1  
0                                                                                                                                                                                                                       Country interviewee is in.  
1                                                                                                                                                                                                                         Year survey was done in.  
2                                                                                                                                                                                                           Unique identifier for each interviewee  
3                                                                                                                                                                                                                   Type of location: Rural, Urban  
4                                                                                                                                                                                                If interviewee has access to a cellphone: Yes, No  
5                                                                                                                                                                                                             Number of people living in one house  
6                                                                                                                                                                                                                       The age of the interviewee  
7                                                                                                                                                                                                              Gender of interviewee: Male, Female  
8                                                                                               The interviewee’s relationship with the head of the house:Head of Household, Spouse, Child, Parent, Other relative, Other non-relatives, Dont know  
9                                                                                                                    The martial status of the interviewee: Married/Living together, Divorced/Seperated, Widowed, Single/Never Married, Don’t know  
10                                                                               Highest level of education: No formal education, Primary education, Secondary education, Vocational/Specialised training, Tertiary education, Other/Dont know/RTA  
11  Type of job interviewee has: Farming and Fishing, Self employed, Formally employed Government, Formally employed Private, Informally employed, Remittance Dependent, Government Dependent, Other Income, No Income, Dont Know/Refuse to answer

In [61]:
class DataUnderstanding:
	'''
	This is a class that will handle all the data understanding process;
	The data info, shape of the data, checking for missing values, 
	checking for duplications in the data and data overview
	'''
	def __init__(self,df):
		self.df = df
		self.shape = self.df.shape
		self.info = self.df.info
		self.describe = self.df.describe
		self.head = self.df.head(5)
		self.duplicates = self.df.duplicated().sum()
		self.missing = self.df.isna().sum()

In [87]:
# Initialize the data understanding class
train_df = DataUnderstanding(train)
test_df = DataUnderstanding(test)

Data Overview

In [88]:
train_df.head

country  year    uniqueid bank_account location_type cellphone_access  \
0   Kenya  2018  uniqueid_1          Yes         Rural              Yes   
1   Kenya  2018  uniqueid_2           No         Rural               No   
2   Kenya  2018  uniqueid_3          Yes         Urban              Yes   
3   Kenya  2018  uniqueid_4           No         Rural              Yes   
4   Kenya  2018  uniqueid_5           No         Urban               No   

   household_size  age_of_respondent gender_of_respondent  \
0               3                 24               Female   
1               5                 70               Female   
2               5                 26                 Male   
3               5                 34               Female   
4               8                 26                 Male   

  relationship_with_head           marital_status  \
0                 Spouse  Married/Living together   
1      Head of Household                  Widowed   
2         Other relative     Single/Never Married   
3      Head of Household  Married/Living together   
4                  Child     Single/Never Married   

                   education_level                   job_type  
0              Secondary education              Self employed  
1              No formal education       Government Dependent  
2  Vocational/Specialised training              Self employed  
3                Primary education  Formally employed Private  
4                Primary education        Informally employed

In [89]:
test_df.head

country  year       uniqueid location_type cellphone_access  household_size  \
0   Kenya  2018  uniqueid_6056         Urban              Yes               3   
1   Kenya  2018  uniqueid_6060         Urban              Yes               7   
2   Kenya  2018  uniqueid_6065         Rural               No               3   
3   Kenya  2018  uniqueid_6072         Rural               No               6   
4   Kenya  2018  uniqueid_6073         Urban               No               3   

   age_of_respondent gender_of_respondent relationship_with_head  \
0                 30                 Male      Head of Household   
1                 51                 Male      Head of Household   
2                 77               Female                 Parent   
3                 39               Female      Head of Household   
4                 16                 Male                  Child   

            marital_status                  education_level  \
0  Married/Living together              Secondary education   
1  Married/Living together  Vocational/Specialised training   
2  Married/Living together              No formal education   
3  Married/Living together                Primary education   
4     Single/Never Married              Secondary education   

                       job_type  
0  Formally employed Government  
1     Formally employed Private  
2          Remittance Dependent  
3          Remittance Dependent  
4          Remittance Dependent

In [90]:
# Shape of the data
print(f'''
Shape of train df is {train_df.shape} with {train_df.shape[0]} rows and {train_df.shape[1]} columns
Shape of test df is {test_df.shape} with {test_df.shape[0]} rows and {test_df.shape[1]} columns
''')


Shape of train df is (23524, 13) with 23524 rows and 13 columns
Shape of test df is (10086, 12) with 10086 rows and 12 columns



General Information about the data

In [91]:
train_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 23524 entries, 0 to 23523
Data columns (total 13 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   country                 23524 non-null  object
 1   year                    23524 non-null  int64 
 2   uniqueid                23524 non-null  object
 3   bank_account            23524 non-null  object
 4   location_type           23524 non-null  object
 5   cellphone_access        23524 non-null  object
 6   household_size          23524 non-null  int64 
 7   age_of_respondent       23524 non-null  int64 
 8   gender_of_respondent    23524 non-null  object
 9   relationship_with_head  23524 non-null  object
 10  marital_status          23524 non-null  object
 11  education_level         23524 non-null  object
 12  job_type                23524 non-null  object
dtypes: int64(3), object(10)
memory usage: 2.3+ MB


In [92]:
test_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10086 entries, 0 to 10085
Data columns (total 12 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   country                 10086 non-null  object
 1   year                    10086 non-null  int64 
 2   uniqueid                10086 non-null  object
 3   location_type           10086 non-null  object
 4   cellphone_access        10086 non-null  object
 5   household_size          10086 non-null  int64 
 6   age_of_respondent       10086 non-null  int64 
 7   gender_of_respondent    10086 non-null  object
 8   relationship_with_head  10086 non-null  object
 9   marital_status          10086 non-null  object
 10  education_level         10086 non-null  object
 11  job_type                10086 non-null  object
dtypes: int64(3), object(9)
memory usage: 945.7+ KB


Check for missing data

In [93]:
train_df.missing

country                   0
year                      0
uniqueid                  0
bank_account              0
location_type             0
cellphone_access          0
household_size            0
age_of_respondent         0
gender_of_respondent      0
relationship_with_head    0
marital_status            0
education_level           0
job_type                  0
dtype: int64

In [94]:
test_df.missing

country                   0
year                      0
uniqueid                  0
location_type             0
cellphone_access          0
household_size            0
age_of_respondent         0
gender_of_respondent      0
relationship_with_head    0
marital_status            0
education_level           0
job_type                  0
dtype: int64

Check for duplicates in the data

In [97]:
print(f'''
Train df: There are {train_df.duplicates} duplicated records in the dataframe
Test df: There are {test_df.duplicates} duplicated records in the dataframe 
''')


Train df: There are 0 duplicated records in the dataframe
Test df: There are 0 duplicated records in the dataframe 



Summary Statistics

In [99]:
train_df.describe()

year  household_size  age_of_respondent
count  23524.000000    23524.000000       23524.000000
mean    2016.975939        3.797483          38.805220
std        0.847371        2.227613          16.520569
min     2016.000000        1.000000          16.000000
25%     2016.000000        2.000000          26.000000
50%     2017.000000        3.000000          35.000000
75%     2018.000000        5.000000          49.000000
max     2018.000000       21.000000         100.000000

In [100]:
test_df.describe()

year  household_size  age_of_respondent
count  10086.000000    10086.000000       10086.000000
mean    2016.975907        3.778802          38.308348
std        0.847356        2.212721          16.270053
min     2016.000000        1.000000          16.000000
25%     2016.000000        2.000000          26.000000
50%     2017.000000        3.000000          35.000000
75%     2018.000000        5.000000          48.000000
max     2018.000000       20.000000         100.000000

Summary of the Data Understanding:
- The train data:
    - There are no missing or duplicated records in the data.
    - There are 3 numerical and 9 categorical columns
    - It consists of 23524 rows and 13 columns
- The test data
    - There are no missing or duplicated records in the data.
    - There are 3 numerical and 9 categorical columns
    - It consists of 10086 rows and 12 columns

## Data Preparation

EDA